spinalcordtoolbox/reports/qc.py

```diff
@@ -661,9 +663,9 @@ def generate_qc(fname_in1, fname_in2=None, fname_seg=None, plane=None, args=None
     """
     logger.info('\n*** Generate Quality Control (QC) html report ***')
     dpi = 300  # Output resolution of the image
-    qcslice_type = None
-    qcslice_operations = None
-    qcslice_layout = None
+    qcslice_type: qcslice.Slice
+    qcslice_operations: List[Callable[[QcImage, np.ndarray, Axes], None]]
+    qcslice_layout: Callable[[qcslice.Slice], (List[np.ndarray] | Tuple[List[List[np.ndarray]], List[Tuple[int, int]]])]
```

Comment:
Good point, I'll add a comment.

Re: PyCharm's complaint, it looks like a few things are happening. Some of it is our fault, some of it is PyCharm's fault:
- Because the file contains almost no type annotations, PyCharm has to guess the types of most things automatically.
- In particular, because `QcImage.label_vertebrae` uses the construct `mask < 1` ([here](https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/52c486fc2b5c04a64f663405c2d94947aec93520/spinalcordtoolbox/reports/qc.py#LL136C3-L136C3)), PyCharm infers that this function needs a `mask` object which implements the `__lt__` special method. This is true.
- However, PyCharm seems to also have replaced `mask: np.ndarray` from the type hint with `mask: Any`, which is... not true? But probably the kind of simplification that makes PyCharm's type inference work reasonably well in practice.
- In contrast, `QcImage.grid` only uses the `mask` parameter completely generically, so that's probably why PyCharm has no complaint there.
- Another source of weirdness is that all of these `QcImage.something` functions are defined as methods, but actually used as functions (subtle distinction). And here, it looks like PyCharm's type inference thinks that `QcImage.label_vertebrae` is a bound method, when in fact it's an unbound function. To illustrate:

  ```py3
  >>> # Create a class with a method
  >>> class A:
  ...     def f(self): return 3
  ... 
  >>> # Create an instance of this class
  >>> a = A()
  >>> 
  >>> # Note that `A.f` and `a.f` are not the same kind of thing:
  >>> A.f
  <function A.f at 0x7f50b7543ee0>
  >>> a.f
  <bound method A.f of <__main__.A object at 0x7f50b7965040>>
  >>> 
  >>> # In particular, you have to manually supply the `self` when calling `A.f`:
  >>> A.f(a)
  3
  >>> 
  >>> # But as a bound method, you **don't** supply `self` as an argument:
  >>> a.f()
  3
  ```

I don't use PyCharm myself, so it's harder for me to test this, but I guess I could install it and try a few solutions. In particular, maybe putting some more type annotations around would help.

Reply:
Honestly, I could always turn off these specific warnings on my end, too? I don't want to put you through all that work just for some yellow highlighting, especially when you've already done quite a bit of investigation just for this one comment.